content/articles/a-look-at-c#-coding-conventions/index.md

```diff
@@ -0,0 +1,238 @@
+
+```c#
+public interface IPublicInstitution
+{
+}
+```
+
+Besides to what it's already there, public members of types can be named using the pascal case.
+
+```c#
+public class PracticeExample1
+{
+    // public field
+    public bool IsValid;
+
+    // An init-only property is a property that allows you to do something only once.
+    public IBuilding Building { get; init; }
+
+    // An example of an event
+    public event Action EventProcessing;
+
+    // Implementation of a method
+    public void StartEventProcessing()
+    {
+        // Use of a local function
+        static int CountQueueItems() => WorkerQueue.Count;
+
+    }
+}
+```
+
+Since parameters are public properties of the record, a reader can use pascal casing when putting down positional accounts.
+
+```c#
+public record PublicIntitution
+(
+    string Country,
+    string County,
+    string Village,
+);
+```
+
+### Camel case
+The camelCasing convention, which is used for parameter names, capitalizes the first letter of each word except the first.
+
+```
+voidMain
+```
+
+Another implementation of the camel case is when naming private or internal fields.
+
+```
+camelCasing
+```
+
+```c#
+public class DataService
+{
+    private IWorkerQueue _workerQueue;
+}
+```
+
+It is also advisable to use the `s_prefix` and the `t_prefix` for the static fields respectively when dealing with thread and internal fields.
+
+```c#
+public class DataService
+{
+    private static IWorkerQueue s_workerQueue;
+
+    [ThreadStatic]
+    private static TimeSpan t_timeSpan;
+}
+```
+
+Camel casing is used when executing most of the method parameters.
+
+```c#
+public T SomeMethod<T>(int someNumber, bool isValid)
+{
+}
+```
+
+Apart from the above conventions, there are other several conventions that a programmer needs to consider and apply while in their daily routine of programming.
+1. The choice of word - Use readable identifier names. Underscores, hyphens, and other non-alphanumeric characters should not be used.
+2. Hungarian notation is neither acceptable nor encouraged.
+3. The use of Abbreviations and Acronyms - Do not use acronyms that aren't generally recognized, and even if they are, only use them when essential.
+4. Abbreviations and contractions should not be used as identifier names.
+5. Use `GetWindow` instead of `GetWin`.
+
+### Layout Conventions
+For a clear understanding of the code and its structure, formatting is exercised as a coding convention.
+
+Some of the layout coding conventions are:
+- Use the code editor's default settings that use smart indenting, four-character indents, and tabs that are saved as spaces.
+- Each line should accommodate one statement.
+- Each line should only accommodate one declaration.
+- If the continuation lines are not indented consequently, they are supposed to be indented at a stop of one tab.
+- At least a single blank line is supposed to be added between the property definitions and the method.
```

Comment:
Explain why this is supposed to be done

Reply:
All this is done not that it is too important but from what the experience I have gained and advice from my seniors, it is done to make sure that your code is clean and neat so that it is readable, a reader will have an easy time following your code and to know what each line entails.